core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +707,86 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  /*
+   * Puts ClassHierarchy into a JSON readable variable that maps the class to a list of its
+   * subclasses
+   */
+  public String toJson() {
+    // initialize variables (subclass and dag) to store important values like hashmap of each
+    // iteration and a list of all maps of class to subclasses
+    HashMap<String, Set<String>> subclass = new HashMap<>();
+    Iterator<Node> children = root.getChildren();
+    // Goes through all subclasses of the root class, and adds the subclass hashmap to the
+    // list of all subclass hashmaps
+    while (children.hasNext()) {
+      helperToJson(children.next(), subclass);
+    }
+    // Removes unnecesarry parts from name of the class
+    String key = nodeToString(root);
+    // inserting the root class to its subclasses into the main hashmap while removing
+    // unnecessary substrings from the class name
+    Iterator<Node> rootChildren = root.getChildren();
+    Set<String> finalRootChildren = new HashSet<>();
+    while (rootChildren.hasNext()) {
+      finalRootChildren.add(nodeToString(rootChildren.next()));
+    }
+    subclass.put(key, finalRootChildren);
+    Gson gson = new Gson();
+    return gson.toJson(subclass);
+  }
+
+  /*
+   * helper function to toJson that performs recursion to go through all of the DAG
+   */
+  private void helperToJson(Node n, HashMap<String, Set<String>> hash) {
+    if (n.children.size() > 0) {
+      // while loop: defines iterator to go through the class's subclasses
+      Iterator<Node> children = n.getChildren();
+      while (children.hasNext()) {
+        // puts subclass variable into temp, and start recursion from that subclass
+        // until base case reaches
+        Node temp = children.next();
+        helperToJson(temp, hash);
+        // remove unnecessary substrings from class name
+        String key = nodeToString(temp);
+        // if statement: only adds the map from class to subclass if class name does
+        // not exist in the overall hashmap
+        if (!hash.containsKey(key)) {
+          // Iterates through the subclass list and removes unnecessary substrings from
+          // class name --> adds that new value into varibale, root, containig all it's subclasses
+          Iterator<Node> val = temp.getChildren();
+          Set<String> root = new HashSet<>();
+          while (val.hasNext()) {
+            Node tempVal = val.next();
+            String newVal = nodeToString(tempVal);
+            root.add(newVal);
+          }
+          // insert into hashmap containing all class to subclass pairs
+          hash.put(key, root);
```

Comment:
This code pattern is duplicated also in the `toJSON` method.  Can we extract the code to a single method and call that method in both places?